presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -768,6 +772,14 @@ public void addTableColumnReferences(AccessControl accessControl, Identity ident
                 .forEach((key, value) -> references.computeIfAbsent(key, k -> new HashSet<>()).addAll(value));
     }
 
+    public void addTableColumnWithSubFieldReferences(AccessControl accessControl, Identity identity, Multimap<QualifiedObjectName, String> tableColumnMap)
```

Comment:
Do we need to create a similar function as the one above? Or we can combine them and get all the information with one pass?